In [ ]:
import pandas as ps
import os
import shutil

In [ ]:
# collecting data for positve patients form source1 (github)

dataPath= "covid-chestxray-dataset-master/metadata.csv"
rawImgsPath= "covid-chestxray-dataset-master/images"


In [ ]:
# discoving the dataset

df = ps.read_csv(dataPath)
print (df.shape)

(950, 30)


so we have 950 rows and 30 columns 

In [ ]:
df.head(5)

patientid  offset sex   age                   finding RT_PCR_positive  \
0         2     0.0   M  65.0  Pneumonia/Viral/COVID-19               Y   
1         2     3.0   M  65.0  Pneumonia/Viral/COVID-19               Y   
2         2     5.0   M  65.0  Pneumonia/Viral/COVID-19               Y   
3         2     6.0   M  65.0  Pneumonia/Viral/COVID-19               Y   
4         4     0.0   F  52.0  Pneumonia/Viral/COVID-19               Y   

  survival intubated intubation_present went_icu  ...              date  \
0        Y         N                  N        N  ...  January 22, 2020   
1        Y         N                  N        N  ...  January 25, 2020   
2        Y         N                  N        N  ...  January 27, 2020   
3        Y         N                  N        N  ...  January 28, 2020   
4      NaN         N                  N        N  ...  January 25, 2020   

                                            location  folder  \
0        Cho Ray Hospital, Ho Chi Minh City, Vietnam  images   
1        Cho Ray Hospital, Ho Chi Minh City, Vietnam  images   
2        Cho Ray Hospital, Ho Chi Minh City, Vietnam  images   
3        Cho Ray Hospital, Ho Chi Minh City, Vietnam  images   
4  Changhua Christian Hospital, Changhua City, Ta...  images   

                                            filename                   doi  \
0  auntminnie-a-2020_01_28_23_51_6665_2020_01_28_...  10.1056/nejmc2001272   
1  auntminnie-b-2020_01_28_23_51_6665_2020_01_28_...  10.1056/nejmc2001272   
2  auntminnie-c-2020_01_28_23_51_6665_2020_01_28_...  10.1056/nejmc2001272   
3  auntminnie-d-2020_01_28_23_51_6665_2020_01_28_...  10.1056/nejmc2001272   
4                              nejmc2001573_f1a.jpeg  10.1056/NEJMc2001573   

                                                 url  license  \
0  https://www.nejm.org/doi/full/10.1056/NEJMc200...      NaN   
1  https://www.nejm.org/doi/full/10.1056/NEJMc200...      NaN   
2  https://www.nejm.org/doi/full/10.1056/NEJMc200...      NaN   
3  https://www.nejm.org/doi/full/10.1056/NEJMc200...      NaN   
4  https://www.nejm.org/doi/full/10.1056/NEJMc200...      NaN   

                                      clinical_notes other_notes Unnamed: 29  
0  On January 22, 2020, a 65-year-old man with a ...         NaN         NaN  
1  On January 22, 2020, a 65-year-old man with a ...         NaN         NaN  
2  On January 22, 2020, a 65-year-old man with a ...         NaN         NaN  
3  On January 22, 2020, a 65-year-old man with a ...         NaN         NaN  
4   diffuse infiltrates in the bilateral lower lungs         NaN         NaN  

[5 rows x 30 columns]

Data shows that we have data of positve and negative corona patiants, we need to extract the positive ones

In [ ]:
covidDir = "covidData/covid"

if not os.path.exists(covidDir):
    os.mkdir(covidDir)
    print("covid folder created")

covid folder created


In [ ]:
counter = 0

for (i,row) in df.iterrows():
    if row["finding"]== "Pneumonia/Viral/COVID-19":
        counter+=1
        
print(counter)        
        

584


so we have 584 positve case in the dataset provided form the github repo

In [ ]:
df["view"]

0      PA
1      PA
2      PA
3      PA
4      PA
       ..
945    AP
946    AP
947    PA
948    AP
949     L
Name: view, Length: 950, dtype: object

going throgh the following link, I found that there is many views for a single X-Ray the pa is the frontend of it and what we baisclly looking for
https://en.wikipedia.org/wiki/Chest_radiograph

In [ ]:
counter = 0

for (i,row) in df.iterrows():
    if row["finding"]== "Pneumonia/Viral/COVID-19" and row["view"]=="PA":
        imageName = row["filename"]
        imagePath = os.path.join(rawImgsPath,imageName)
        imgCopyPath = os.path.join(covidDir,imageName)
        shutil.copy2(imagePath,imgCopyPath)
        print("Transfar image",counter)
        counter+=1
        
print(counter) 

Transfar image 0
Transfar image 1
Transfar image 2
Transfar image 3
Transfar image 4
Transfar image 5
Transfar image 6
Transfar image 7
Transfar image 8
Transfar image 9
Transfar image 10
Transfar image 11
Transfar image 12
Transfar image 13
Transfar image 14
Transfar image 15
Transfar image 16
Transfar image 17
Transfar image 18
Transfar image 19
Transfar image 20
Transfar image 21
Transfar image 22
Transfar image 23
Transfar image 24
Transfar image 25
Transfar image 26
Transfar image 27
Transfar image 28
Transfar image 29
Transfar image 30
Transfar image 31
Transfar image 32
Transfar image 33
Transfar image 34
Transfar image 35
Transfar image 36
Transfar image 37
Transfar image 38
Transfar image 39
Transfar image 40
Transfar image 41
Transfar image 42
Transfar image 43
Transfar image 44
Transfar image 45
Transfar image 46
Transfar image 47
Transfar image 48
Transfar image 49
Transfar image 50
Transfar image 51
Transfar image 52
Transfar image 53
Transfar image 54
Transfar image 55
Tr

so we got about 196 pic of a frontend chest x-rays

In [ ]:
# collecting some of the non-covid images form the source 2 (kaggle)

import random

kagglePath = "chest_xray kaggle/train/NORMAL"
nonCovidDir = "covidData/nonCovid"

imgNames = os.listdir(kagglePath)
random.shuffle (imgNames)


In [ ]:
nonCovidDir = "covidData/nonCovid"

if not os.path.exists(nonCovidDir):
    os.mkdir(nonCovidDir)
    print("covid folder created")



for i in range (196):
    imgName = imgNames[i]
    imgPath = os.path.join(kagglePath,imgName)
    imgCopyPath = os.path.join(nonCovidDir,imgName)
    shutil.copy2(imgPath,imgCopyPath)
    print("Transfar image",i)

covid folder created
Transfar image 0
Transfar image 1
Transfar image 2
Transfar image 3
Transfar image 4
Transfar image 5
Transfar image 6
Transfar image 7
Transfar image 8
Transfar image 9
Transfar image 10
Transfar image 11
Transfar image 12
Transfar image 13
Transfar image 14
Transfar image 15
Transfar image 16
Transfar image 17
Transfar image 18
Transfar image 19
Transfar image 20
Transfar image 21
Transfar image 22
Transfar image 23
Transfar image 24
Transfar image 25
Transfar image 26
Transfar image 27
Transfar image 28
Transfar image 29
Transfar image 30
Transfar image 31
Transfar image 32
Transfar image 33
Transfar image 34
Transfar image 35
Transfar image 36
Transfar image 37
Transfar image 38
Transfar image 39
Transfar image 40
Transfar image 41
Transfar image 42
Transfar image 43
Transfar image 44
Transfar image 45
Transfar image 46
Transfar image 47
Transfar image 48
Transfar image 49
Transfar image 50
Transfar image 51
Transfar image 52
Transfar image 53
Transfar image 54